renderdoc/driver/vulkan/vk_acceleration_structure.cpp

```diff
@@ -580,75 +606,36 @@ bool VulkanAccelerationStructureManager::Serialise(SerialiserType &ser, Resource
                                                    CaptureState state)
 {
   VkDevice d = !IsStructuredExporting(state) ? m_pDriver->GetDev() : VK_NULL_HANDLE;
-  const bool replayingAndReading = ser.IsReading() && IsReplayMode(state);
   VkResult vkr = VK_SUCCESS;
 
+  VkAccelerationStructureInfo *asInfo =
+      initial ? initial->accelerationStructureInfo : new VkAccelerationStructureInfo();
   byte *contents = NULL;
-  uint64_t contentsSize = initial ? initial->mem.size : 0;
-  MemoryAllocation mappedMem;
-
-  // Serialise this separately so that it can be used on reading to prepare the upload memory
-  SERIALISE_ELEMENT(contentsSize);
 
   const VkDeviceSize nonCoherentAtomSize = m_pDriver->GetDeviceProps().limits.nonCoherentAtomSize;
+  Allocation uploadMemory;
 
-  // the memory/buffer that we allocated on read, to upload the initial contents.
-  MemoryAllocation uploadMemory;
-  VkBuffer uploadBuf = VK_NULL_HANDLE;
+  SERIALISE_ELEMENT(*asInfo).Hidden();
+  RDCASSERT(asInfo);
```

Comment:
This assert would be better before the serialise as otherwise if it's ever NULL we will crash before getting to the assert.